docs/user-guide/examples.md

```diff
@@ -130,4 +130,47 @@ defaultEntryPoints = ["http"]
     headerField = "X-WebAuth-User"
     [entryPoints.http.auth.basic]
     users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
-```
\ No newline at end of file
+```
+
+
+## Enable authentication forwarding
+
+Authentication forwarding is a mechanism to allow for arbitrary authentication back ends to be used.
+It works by forwarding any incoming request from traefik proxy to a specified endpoint (the setting key `entryPoints.http.auth.forward.address`).
+This endpoint address is called "authentication back end".
+The response of the authentication back end is then evaluated.
+The authentication back end has to send a response with an HTTP status code 200 to allow access.
+Any other status code will result in traefik returning the authentiucation back end response to the original request (so if your back end sends a response with a 503 header and any body, traefik will also send a 503 response including the body).
+
+In addition to this, traefik provides basic functionality to modify the original request's query parameters before they are sent to the authentication back end.
+This allows you to map a request parameter or header (in this case token) from e.g. `traefik.com/secret?token=foobar` to `authserver.com/auth?theToken=foobar`.
```

Comment:
I'm -1 on this, it seems specific for a particular implementation of an auth backend. 

I think the best approach is to just forward the request and let the backend extract what it needs (eg the functionality offered by `entryPoints.http.auth.forward.forwardAllHeaders=true`).